api/rosetta/balance.go

```diff
@@ -56,14 +56,25 @@ func (d *Data) Balance(ctx echo.Context) error {
 		return echo.NewHTTPError(http.StatusBadRequest, InvalidFormat("block identifier: at least one of hash or index is required"))
 	}
 	if req.BlockID.Hash != "" && len(req.BlockID.Hash) != hexIDSize {
-		return echo.NewHTTPError(http.StatusBadRequest, InvalidFormat("block identifier: hash field has wrong length (have: %d, want: %d)", len(req.BlockID.Hash), hexIDSize))
+		return echo.NewHTTPError(
+			http.StatusBadRequest,
```

Comment:
The more I read the rosetta documentation the more my interpretation changes.

The docs state that

> (description) MUST NOT be populated with information about a particular instantiation of an error (use details for this).

Does this include messages like `blockchain field is empty` and `hash field has wrong length`? This is also information about a specific instance of the error (all falling under `ErrorInvalidFormat`).

If so, this should also be moved to the `details` field. Then we might as well have the description JSON field be `omitempty` and omit it if it's going to be that slim with information..

Reply:
Yeah, that's what I thought too. Unfortunately the examples in their docs are pretty monotone and use the same errors all the time basically. And in their examples, the descriptions are indeed a bit useless.

e.g.
```json
{
    "code": 12,
    "message": "Invalid account format",
    "description": "This error is returned when the requested AccountIdentifier is improperly formatted.",
    "retriable": true,
    "details": {
        "address": "0x1dcc4de8dec75d7aab85b567b6",
        "error": "not base64"
    }
}
```

In the docs they also state:

> Description allows the implementer to optionally provide additional information about an error. In many cases, the content of this field will be a copy-and-paste from existing developer documentation. Description can ONLY be populated with generic information about a particular type of error.

Reply:
I think your interpretation is correct, and the description is meant to always be the same for the same code/message. However, it also says that the description can change between _versions_ of the implementation. As this is a field meant for human consumption, I say that we simply take the freedom to use different messages here, depending on the sub-class of the error.